__tests__/app.test.js

```diff
@@ -298,7 +298,67 @@ describe("GET/api/articles/:article_id/comments", () => {
       .get("/api/articles/9999/comments")
       .expect(404)
       .then(({ body }) => {
-        expect(body.msg).toBe("Route Not Found");
+        expect(body.msg).toBe("Route not Found");
+      });
+  });
+});
+
+const newComment = { username: "lurker", body: "This is a very good article" };
+
+describe("POST/api/articles/:article_id/comments", () => {
+  test("Responds with the posted comment ", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        const { comment } = body;
+        expect(comment).toEqual(
+          expect.objectContaining({
+            comment_id: 19,
+            body: "This is a very good article",
+            votes: 0,
+            author: "lurker",
+            article_id: 1,
+            created_at: expect.any(String),
+          })
+        );
+      });
+  });
+  test("status 400: Returns a bad request message when given a endpoint of wrong type", () => {
+    return request(app)
+      .post("/api/articles/banana/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test("status 404: Returns a route not found message when given a non existent endpoint ", () => {
+    return request(app)
+      .post("/api/articles/9999/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Route not Found");
+      });
+  });
+  test("status 422: returns an incorrect message when sent invalid data key", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({ leg: "This is a very good article", username: "lurker" })
+      .expect(422)
```

Comment:
Nice job identifying this could be a 422!

I've not actually come across it before but it seems to be a good choice :)